app/models/katello/host/subscription_facet.rb

```diff
@@ -35,6 +39,35 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_hypervisor(consumer_params = candlepin_consumer.consumer_attributes)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.hypervisor = true
+        elsif !consumer_params.try(:[], 'guestIds').empty?
+          self.hypervisor = true
+        elsif !candlepin_consumer.virtual_guests.empty?
+          # Check by calling out to Candlepin last for efficiency
+          self.hypervisor = true
+        end
+      end
+
+      def update_guests(consumer_params)
+        if self.hypervisor
+          if !consumer_params.try(:[], 'guestIds').empty?
+            guest_ids = consumer_params['guestIds'].map{ |i| i['guestId'].downcase}
```

Comment:
Try an older version of virt-who. Previously the json sent up was literally just: $hypervisor: [$id1, $id2, $id3]

Since this is our API I don't think we should change the format of a parameter. Perhaps accept both new and old and convert at the controller?